### ISSUE TITLE
`bush up folder/` is broken

### ISSUE THREAD

**Issue by wapiflapi** [BestPig/bush#25]
Trying to upload a folder causes bush to exit with an error status:

``` py3tb
Traceback (most recent call last):
  File "/usr/local/bin/bush", line 12, in <module>
    exec(compile(open(__file__).read(), __file__, 'exec'))
  File "/home/wapiflapi/Projects/bush/client/bin/bush", line 6, in <module>
    bush.cli.main()
  File "/home/wapiflapi/Projects/bush/client/bush/cli.py", line 128, in main
    args.callback(api, args)
  File "/home/wapiflapi/Projects/bush/client/bush/cli.py", line 58, in do_upload_file
    api.upload(args.file, tag=args.tag, callback=ShowProgress)
  File "/home/wapiflapi/Projects/bush/client/bush/api.py", line 75, in upload
    tag = tag or self.tag_for_path(filepath)
  File "/home/wapiflapi/Projects/bush/client/bush/api.py", line 36, in tag_for_path
    return next(part for part in basename.split('.') if part)
StopIteration
```

**Comment by wapiflapi** [comment]
the problem is the tag is deduced from the basename which is the empty string for a path like `'/foo/bar/'`. This is easy to fix with:

``` diff
diff --git a/client/bush/api.py b/client/bush/api.py
index 9c6617f..846f1ed 100644
--- a/client/bush/api.py
+++ b/client/bush/api.py
@@ -32,7 +32,7 @@ class BushAPI():
         return urllib.parse.urljoin(self.base, url)

     def tag_for_path(self, filepath):
-        basename = os.path.basename(filepath)
+        basename = os.path.basename(os.path.realpath(filepath))
         return next(part for part in basename.split('.') if part)

     def sanitize_tag(self, tag):
```

However this raises another/different issue: The server returns this response:

```
--------------------------------------------------------------------------------
<Response [200]>
body content:

--------------------------------------------------------------------------------
```

Which causes the following error when trying to check the `'status'` in the JSON the server is supposed to send back. Do you have an idea what could cause this @BestPig ?

``` py3tb
Traceback (most recent call last):
  File "/usr/local/bin/bush", line 12, in <module>
    exec(compile(open(__file__).read(), __file__, 'exec'))
  File "/home/wapiflapi/Projects/bush/client/bin/bush", line 6, in <module>
    bush.cli.main()
  File "/home/wapiflapi/Projects/bush/client/bush/cli.py", line 128, in main
    args.callback(api, args)
  File "/home/wapiflapi/Projects/bush/client/bush/cli.py", line 58, in do_upload_file
    api.upload(args.file, tag=args.tag, callback=ShowProgress)
  File "/home/wapiflapi/Projects/bush/client/bush/api.py", line 117, in upload
    data = r.json()
  File "/usr/local/lib/python3.4/dist-packages/requests/models.py", line 776, in json
    return json.loads(self.text, **kwargs)
  File "/usr/local/lib/python3.4/dist-packages/simplejson/__init__.py", line 501, in loads
    return _default_decoder.decode(s)
  File "/usr/local/lib/python3.4/dist-packages/simplejson/decoder.py", line 370, in decode
    obj, end = self.raw_decode(s)
  File "/usr/local/lib/python3.4/dist-packages/simplejson/decoder.py", line 393, in raw_decode
    return self.scan_once(s, idx=_w(s, idx).end())
simplejson.scanner.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

**Comment by wapiflapi** [comment]
Still not sure why the server fails in this case, but it is caused by the client sending a '/' in the name. This is fixed in #26. Maybe we should create a separate issue for the server's behavior?
